src/acl/Address.h

```diff
@@ -28,6 +29,7 @@ class Address
     ACLList *aclList;
 
     Ip::Address addr;
+    String label;
```

Comment:
Please do not add such a large/heavy/deprecated data member as String to Acl::Address, especially when the code is using that member for an (essentially) boolean test.

We should also protect the code using Acl::Address from accidentally accessing "unset" `addr`. The current PR  already introduced a bug where an unset `addr` is used, and there will be more such bugs if we do not act now.

For this PR, using std::optional<Ip::Address> instead of Ip::Address may be sufficient -- the code would interpret the no-address state as a "transparent" state. However, using that approach essentially offloads the heavy lifting to the next developer who needs to add a second "special value" to Acl::Address.

```suggestion
    /// specific IP address (when set) or "use client address" prescription (otherwise)
    std::optional<Ip::Address>;
```


If you want to go for a more future-compatible solution than std::optional, then the best solution for problems like this is probably [std::variant](https://en.cppreference.com/w/cpp/utility/variant) (i.e. a type-safe union). The two existing variants (in this PR) are Ip::Address and UseClientAddress. Here is a declaration sketch:

```suggestion
    struct UseClientAddress {};
    using AddressSource = std::variant<Ip::Address, UseClientAddress>;
    AddressSource addressSource; ///< Ip::Address provider
```

Disclaimer: All suggestions above are untested sketches. All may need adjustments and all do require code changes not shown in the suggestions themselves.

Reply:
So this is the first time that `std::variant` is being used in this project, and my first time implementing a union pattern like this. At present, I don't think this approach bought us anything compared to the `std::optional` approach (or even a simple bool/int) - and as such, I'm not really confident I implemented this in the proper way. But as you said, the next dev to work on `tcp_outgoing_address` will (probably) find having this useful.